### PR TITLE
Di 19692 add pfs endpoints to papi sdk

### PR DESCRIPTION
**Pull request by saargt**

### CODE REVIEW

File: endpoints/addons.ts [hunk 1]
```diff
@@ -167,4 +168,24 @@ export class AddonEndpoint extends Endpoint<Addon> {
...
         relations: new Endpoint<Relation>(this.service, '/addons/data/relations'),
     };
+
+    files = {
+        uuid: (addonUUID: string) => {
+            return {
+                get: async (keyName: string): Promise<AddonFile> => {
+                    return await this.service.get(`/addons/files/${addonUUID}/${keyName}`);
+                },
+                getFiles: async (folder: string, params: FindOptions): Promise<AddonFile[]> => {
```
Comment (Chasky-H) [inline endpoints/addons.ts:178]: this should be the get function.

one param of type: `FindOptions & { folder: string }`

File: endpoints/addons.ts [hunk 2]
```diff
@@ -167,4 +168,24 @@ export class AddonEndpoint extends Endpoint<Addon> {
...
+                    let url = `/addons/files/${addonUUID}`;
+                    const query = Endpoint.encodeQueryParams(params);
+                    url = `${url}?folder=${folder}${query ? `&${query}` : ''}`;
```
Comment (Chasky-H) [inline endpoints/addons.ts:181]: You can do this:
```
Endpoint.encodeQueryParams({
   folder: folder,
   ...params
})

File: entities/addons.ts [hunk 3]
```diff
@@ -100,3 +100,22 @@ export const SchemeFieldTypes = [
...
+            Size: '200x200';
+            URL?: string;
+        },
+    ];
+    Sync?: 'None' | 'Device' | 'DeviceThumbnail' | 'Always';
+    URL?: string;
+    URI?: string;
+    PresignedURL?: string;
+    FileVersion?: string;
+    [key: string]: any;
```
Comment (Chasky-H) [inline entities/addons.ts:120]: You allow addons to save additional data on this object?

File: endpoints/addons.ts (hunk 1 above)
Comment (saargt) [reply endpoints/addons.ts]: Done as per our discussion

File: entities/addons.ts (hunk 3 above)
Comment (saargt) [reply entities/addons.ts]: No. Fixed.

**Review by ShirMarciano — APPROVED** [review]
@Chasky-H  Can I merged it?

File: endpoints/addons.ts (hunk 1 above)
Comment (Chasky-H) [reply endpoints/addons.ts]: The get function should be key just like there with a get and post.
See how this was done in the TableEndpoint.

```
papiClient.addons.files.uuid().key(filePath).get()
```

Comment (saargt) [reply endpoints/addons.ts]: It does. This comment is outdated. Take a look at the changed files.

Comment (saargt) [reply endpoints/addons.ts]: @ShirMarciano pointed out I didn't understand what you meant. In any case, I pushed another commit to attend to this comment.

**Comment by Chasky-H** [comment]
@saargt one last thing...
You need to change the version for the CI/CD to create a new version. Read how in the Readme.

**Pull request merged**